src/storage/s3.rs

```diff
@@ -63,19 +59,31 @@ impl<'a> S3Backend<'a> {
         })
     }
 
-    pub(super) fn store_batch(&mut self, batch: &[Blob]) -> Result<(), Error> {
-        use futures::stream::FuturesUnordered;
-        use futures::stream::Stream;
+    pub(super) fn start_storage_transaction(&self) -> Result<S3StorageTransaction, Error> {
+        Ok(S3StorageTransaction {
+            s3: self,
+            runtime: Runtime::new()?,
```

Comment:
Great idea making it so this is only constructed when uploading, not downloading :)